python_modules/dagster/dagster/_core/definitions/schedule_definition.py

```diff
@@ -242,9 +242,17 @@ def resources(self) -> Resources:
         from dagster._core.execution.build_resources import build_resources
 
         if not self._resources:
+            # Early exit if no resources are defined. This skips unnecessary initialization
+            # entirely. This also fixes an issue where a user had a deployment that depended
+            # on not setting environment variables necessary to access the instance, and then
+            # an upgrade broke them.
```

Comment:
comment nit: i think this could mention the situation we are hoping to guard against (users running gRPC servers that can't load the instance) rather than the specific upgrade bug